aidial_adapter_dial/transformer.py

```
@@ -30,128 +31,121 @@ async def create(
-        content_type: str | None,
-        session: aiohttp.ClientSession,
-    ):
-        content = await src_storage.download(src_url, session)
-        await dest_storage.upload(dest_url, content_type, content, session)
-
-    async def modify_request_attachment(self, attachment: dict) -> None:
-        if "url" not in attachment:
-            return None
-
-        local_url = self.local_storage.to_dial_url(attachment["url"])
-        if local_url is None:
-            return None
+        if remote_path.startswith(f"{self.local_user_bucket}/"):
+            path = remote_path.removeprefix(f"{self.local_user_bucket}/")
+            return f"files/{self.local_user_bucket}/{path}"
+        else:
+            regex = r"appdata/([^/]+)/(.+)"
+            match = re.match(regex, remote_path)
+            if match is None:
+                raise ValueError(
+                    f"The remote file ({remote_url!r}) is expected to be uploaded to a remote appdata path"
+                )
+            _remote_app_name, path = match.groups()
 
-        remote_url = await self.get_remote_url(local_url)
-        content_type = attachment.get("type")
+            return f"files/{self.local_app_data}/{path}"
 
-        async with aiohttp.ClientSession() as session:
-            await self.upload_and_download_file(
+    async def modify_request_attachment(self, attachment: dict) -> None:
+        if (ref_url := attachment.get("reference_url")) and (
+            local_ref_url := self.local_storage.to_dial_url(ref_url)
+        ):
+            try:
+                remote_ref_url = await self.get_remote_url(local_ref_url)
+                attachment["reference_url"] = remote_ref_url
+            except Exception:
+                log.error(f"Failed to get remote URL for {local_ref_url!r}")
+
+        if (url := attachment.get("url")) and (
+            local_url := self.local_storage.to_dial_url(url)
+        ):
+            remote_url = await self.get_remote_url(local_url)
```

Why does `reference_url` wrapped in try/except, but this part doesn't?

---

Indeed. I think we shoudn't shadow the errors and let it fail in **both** cases.